### ISSUE TITLE
Unable to install in Ubuntu 20.04 an nvidia container toolkit with version < 1.14.4

### ISSUE THREAD

**Issue by AlexisGitHu** [NVIDIA/nvidia-container-toolkit#341]
Today I was trying to deploy with nvidia-device-plugin for kubernetes with a rtx 3070 ti. I upgraded my CUDA drivers to 545 and my cuda version to 12.3. 

However, an error triggers and indicates that the version of Cuda drivers is insufficient with the cuda runtime (provided by installing the nvidia-ctk), whose version is 1.14.4.

I guess that by downgrading the container toolkit version will be enough to solve the issue, but the official nvidia links (https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/1.13.1/install-guide.html#install-guide ) install the last version and there is no way to install others.

**Comment by klueska** [comment]
You should also be able to set the envvar `-e NVIDIA_DISABLE_REQUIRE=true` when you start your container. It's not ideal (the check shouldn't fail), but this will bypass the check altogether.

**Comment by elezar** [comment]
@AlexisGitHu the NVIDIA Container Toolkit does not provide a specific CUDA driver version. It is separate from the driver installation. The error you're seeing will present regardless of the NVIDIA Container Toolkit version that you're using.

Adding `NVIDIA_DISABLE_REQUIRE=true` to the device plugin daemonset or setting the `disable-require` option in `/etc/nvidia-container-runtime/config.toml` will ignore the warnings you're getting.

**Comment by AlexisGitHu** [comment]
Alright, let me check it works and I'll close the issue.

Thank you for the velocity of the response!

**Comment by AlexisGitHu** [comment]
I modified the config.toml file and still happens the same error. 

This are all the screenshots related to the issue. 

![Captura de pantalla de 2024-02-07 21-35-44](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/469c142a-6523-4db2-b181-1b303a1e5b66)

![Captura de pantalla de 2024-02-07 21-36-18](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/093938e1-661d-4776-8c8d-06c117128bf1)

![Captura de pantalla de 2024-02-07 21-36-36](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/d8d6dd55-4f2b-4c57-97cb-b95093f1c80a)

![Captura de pantalla de 2024-02-07 21-37-12](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/40b0d619-4cc0-4ebe-a661-dbe679e43e3f)

![Captura de pantalla de 2024-02-07 21-37-45](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/cc50ca08-6d3c-415e-9753-37802e2a7b0a)

![Captura de pantalla de 2024-02-07 21-38-14](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/0ad45011-a22f-4eb0-a2fe-61d95ee2ed67)

The gpu-pod provided is the one given by the following .yaml file:

apiVersion: v1
kind: Pod
metadata:
  name: gpu-pod
spec:
  nodeName: mlops-control-plane
  restartPolicy: Never
  containers:
    - name: cuda-container
      image: nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda10.2
      resources:
        limits:
          nvidia.com/gpu: 1 # requesting 1 GPU
  tolerations:
  - key: nvidia.com/gpu
    operator: Exists
    effect: NoSchedule

the daemonset is the one provided in: https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/v0.14.4/nvidia-device-plugin.yml

**Comment by elezar** [comment]
Could you change the command in the same container to `nvidia-smi` to confirm that that is working correctly?

Is `nvidia` set as the default runtime in your container engine, or are you using a runtime class?

**Comment by AlexisGitHu** [comment]
Nvidia is set as the Default Runtime in docker.

Executing `sudo docker run --rm --gpus all nvidia/cuda:11.6.2-base-ubuntu20.04 nvidia-smi` outputs it correctly:
![Captura de pantalla de 2024-02-07 22-54-15](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/fec12cf7-b169-445e-8338-1a089c9d5a8e)

**Comment by elezar** [comment]
> Nvidia is set as the Default Runtime in docker.

Yes, I rechecked the images that you uploaded and confirmed this. 

With regards to:
> Could you change the command in the same container to nvidia-smi to confirm that that is working correctly?

Here I mean can you set this in the GPU pod that you are launching. So something like:
```
cat <<EOF | kubectl apply -f -
---
apiVersion: v1
kind: Pod
metadata:
  name: gpu-pod
spec:
  runtimeClassName: nvidia
  restartPolicy: OnFailure
  containers:
  - name: cuda-container
    image: nvcr.io/nvidia/k8s/cuda-sample:vectoradd-cuda10.2
    command: ["sleep", "9999"]
    resources:
      limits:
        nvidia.com/gpu: 1
EOF
```
And then exec into the container:
```
kubectl exec -ti gpu-pod -- nvidia-smi
```

**Comment by AlexisGitHu** [comment]
When I tried to `kubectl apply -f` the following error happens with that manifest: Error from server (Forbidden): error when creating "STDIN": pods "gpu-pod" is forbidden: pod rejected: RuntimeClass "nvidia" not found.

I guess that the nvidia plugin needs to be configure such that the RuntimeClass: nvidia is set.

**Comment by elezar** [comment]
@AlexisGitHu sorry. Please remove the runtime class from the spec. I have updated the example above.

**Comment by AlexisGitHu** [comment]
The following error pop ups: 
![Uploading Captura de pantalla de 2024-02-08 18-05-57.png…]()

**Comment by elezar** [comment]
Question: Is kubernetes set up to use Docker or Containerd?

The error seems to indicate that the container runtime is not being invoked, or the device plugin is not making the required modifications to the container.

**Comment by elezar** [comment]
Could you also provide the logs of the device plugin pod? Specifically the nvidia-device-plugin-ctr.

**Comment by AlexisGitHu** [comment]
I do not recall to install cri-dockerd or containerd specifically.

The logs of the nvidia-plugin are the followings:

![Captura de pantalla de 2024-02-08 19-07-00](https://github.com/NVIDIA/nvidia-container-toolkit/assets/56341573/cd399a72-4d55-41e1-b278-b7273d5fd068)
